code/src/sixsq/nuvla/server/util/metadata.clj

```diff
@@ -130,35 +155,40 @@
+  ([child-ns parent-ns spec suffix {:keys [scrud-operations] :as _options}]
    (if-let [parent-ns (as-namespace parent-ns)]
-     (let [child-ns      (as-namespace child-ns)
+     (let [child-ns         (as-namespace child-ns)
+
+           resource-name    (cond-> (ns->type-uri (or child-ns parent-ns))
+                                    suffix (str " \u2014 " suffix))
 
-           resource-name (cond-> (ns->type-uri (or child-ns parent-ns))
-                                 suffix (str " \u2014 " suffix))
+           doc              (get-doc (or child-ns parent-ns))
+           type-uri         (cond-> (ns->type-uri (or child-ns parent-ns))
+                                    suffix (str "-" suffix))
 
-           doc           (get-doc (or child-ns parent-ns))
-           type-uri      (cond-> (ns->type-uri (or child-ns parent-ns))
-                                 suffix (str "-" suffix))
+           common           {:id            "resource-metadata/dummy-id"
+                             :created       "1964-08-25T10:00:00.00Z"
+                             :updated       "1964-08-25T10:00:00.00Z"
+                             :resource-type resource-metadata/resource-type
+                             :acl           (acl-utils/normalize-acl {:owners   ["group/nuvla-admin"]
+                                                                      :view-acl ["group/nuvla-anon"]})
+                             :type-uri      type-uri
+                             :name          resource-name
+                             :description   doc}
 
-           common        {:id            "resource-metadata/dummy-id"
-                          :created       "1964-08-25T10:00:00.00Z"
-                          :updated       "1964-08-25T10:00:00.00Z"
-                          :resource-type resource-metadata/resource-type
-                          :acl           (acl-utils/normalize-acl {:owners   ["group/nuvla-admin"]
-                                                                   :view-acl ["group/nuvla-anon"]})
-                          :type-uri      type-uri
-                          :name          resource-name
-                          :description   doc}
+           attributes       (generate-attributes spec)
 
-           attributes    (generate-attributes spec)
+           scrud-operations (or scrud-operations (get-scrud-operations parent-ns))
 
-           actions       (get-actions parent-ns)
+           actions          (get-actions parent-ns)
 
-           capabilities  (get-capabilities parent-ns)]
+           capabilities     (get-capabilities parent-ns)]
 
        (if (and doc spec type-uri)
-         (cond-> common
+         (cond-> (assoc common :spec spec)
                  attributes (merge attributes)
+                 (seq scrud-operations) (assoc :scrud-operations scrud-operations)
                  (seq actions) (assoc :actions actions)
                  (seq capabilities) (assoc :capabilities capabilities))
```

Comment:
Basic idea is to enrich resource metadata with metainfo about scrud operations and the resource spec. This info will be used for the OpenAPI spec generation, respectively to decide which paths to generate for each resource, and to drive schemas generation.